receivers/discord/discord.go

```diff
@@ -202,50 +201,65 @@ func (d Notifier) SendResolved() bool {
 	return !d.GetDisableResolveMessage()
 }
 
-func (d Notifier) constructAttachments(ctx context.Context, as []*types.Alert, embedQuota int) []discordAttachment {
-	attachments := make([]discordAttachment, 0)
+func (d Notifier) constructAttachments(ctx context.Context, alerts []*types.Alert, embedQuota int) []discordAttachment {
+	attachments := make([]discordAttachment, 0, embedQuota)
+	for _, alert := range alerts {
+		// Check if the image limit has been reached at the start of each iteration.
+		if embedQuota < 1 {
+			break
+		}
 
-	_ = images.WithStoredImages(ctx, d.log, d.images,
-		func(index int, image images.Image) error {
-			if embedQuota < 1 {
-				return images.ErrImagesDone
+		attachment, err := d.getAttachmentFromURL(ctx, alert)
+		if err != nil {
+			if errors.Is(err, images.ErrNoImageForAlert) {
```

Comment:
`errors.Is` is `false` if `err` is nil, so I think this structure can be simplified a bit, and we can get rid of all else cases:
```
attachment, err := ...
if errors.Is(err, images.ErrNoImageForAlert) {
    continue
}
if errors.Is(err, ...) {
    attachment, err := d.getAttachmentFromBytes(...)
    // No continue needed! Note that we're re-using `err` from outer scope. Next check will cover it.. or if this succeeds, it gets set to nil.
    // We can leave the inner `if` if we really want here, but we don't have to.
}
if err != nil {
    d.log.Error(...)
    continue
}
```
Then, it reads more like a book, top-to-bottom, and has less nesting.